enforcer-rules/src/main/java/org/apache/maven/enforcer/rules/dependency/DependencyConvergence.java

```diff
@@ -50,6 +53,8 @@ public final class DependencyConvergence extends AbstractStandardEnforcerRule {
 
     private List<String> excludes;
 
+    private List<String> scopes = Arrays.asList(SCOPE_COMPILE, SCOPE_RUNTIME, SCOPE_IMPORT, SCOPE_SYSTEM);
```

Comment:
It's implied that this can be changed but I don't see where?

Reply:
This will be an input parameter for the rule that can be configured as it is described in the documentation.

Here is an example:
```xml
<dependencyConvergence>
    <scopes>
        <scope>compile</scope>
        <scope>test</scope>
    </scopes>
</dependencyConvergence>
```

Then in the logs you will see:
`[DEBUG] Executing Rule 0: EnforcerRuleDesc[name=dependencyConvergence, rule=DependencyConvergence[includes=null, excludes=null, uniqueVersions=false, scopes=compile,test], level=ERROR]`

Reply:
It is magic from `plexu ComponentConfigurator`, xml from configuration is used to setup fields on rule object instance.